web/server/fastify/server.ts

```diff
@@ -74,29 +74,28 @@ function createH1Server(opts: Options) {
   return fastify(serverOpts);
 }
 
-function createH2Server(opts: Options) {
+export async function start(opts: Options) {
+  const h1Server = createH1Server(opts);
+  await h1Server.register(fastifyCors, interop.corsOptions);
+  await h1Server.register(fastifyConnectPlugin, { routes });
+  await h1Server.listen({ host: HOST, port: opts.h1port });
+
   if (!opts.insecure && opts.key && opts.cert) {
-    return fastify({
+    const h2Server = fastify({
       http2: true,
       https: getTLSConfig(opts.key, opts.cert),
     });
+    await h2Server.register(fastifyCors, interop.corsOptions);
+    await h2Server.register(fastifyConnectPlugin, { routes });
+    await h2Server.listen({ host: HOST, port: opts.h2port });
   } else {
-    return fastify({
+    const h2InsecureServer = fastify({
       http2: true,
     });
+    await h2InsecureServer.register(fastifyCors, interop.corsOptions);
+    await h2InsecureServer.register(fastifyConnectPlugin, { routes });
+    await h2InsecureServer.listen({ host: HOST, port: opts.h2port });
   }
```

Comment:
This is admittedly not great, but not sure of a better way to do this. Something must have changed with Fastify types in some way because with the latest upgrade to `@connect-fastify@v0.13.2` and `fastify@v4.22.0`, it now complains with the following on bother `h2Server.register` lines:

```
server/fastify/server.ts:98:18 - error TS2349: This expression is not callable.
  Each member of the union type 'FastifyRegister<FastifyInstance<Http2Server, Http2ServerRequest, Http2ServerResponse, FastifyBaseLogger, FastifyTypeProviderDefault> & PromiseLike<...>, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger> | FastifyRegister<...>' has signatures, but none of those signatures are compatible with each other.
```

This seems to be because the returned h2Server is a union type but TypeScript doesn't know which type is being used to invoke `register`. Splitting these up allows TypeScript to properly discern the types, but at the expense of code duplication. I tried using a type assertion here, but that didn't seem great either.